engines/e_capi.c

```diff
@@ -541,10 +541,11 @@ static int capi_destroy(ENGINE *e)
 
 static int capi_finish(ENGINE *e)
 {
+    int ret;
     CAPI_CTX *ctx;
     ctx = ENGINE_get_ex_data(e, capi_idx);
     capi_ctx_free(ctx);
-    ENGINE_set_ex_data(e, capi_idx, NULL);
+    ret = ENGINE_set_ex_data(e, capi_idx, NULL);
```

Comment:
this will cause "set but not used" errors on some compilers.  so perhaps ```(void)``` is better.